src/pages/career/[opportunityId].tsx

```diff
@@ -1,19 +1,35 @@
-import { useRouter } from 'next/router';
+import { NextPage, NextPageContext } from 'next';
 import React from 'react';
 
 import DetailView from 'career/containers/DetailView';
-import CareerOpportunities from 'career/providers/CareerProvider';
 
-const CareerDetailPage = () => {
-  const router = useRouter();
-  const opportunityId = Number(router.query.opportunityId);
+import { careerOpportunitySelectors, fetchCareerOpportunityById } from 'career/slices/careerOpportunities';
+import { Store } from 'core/redux/Store';
+
+interface IContext extends NextPageContext {
+  store: Store;
+}
+
+interface IProps {
+  opportunityId: number;
```

Comment:
Just to see if I remember correctly: This is gotten from Next router, right? Based on the id given in the url, which will then match a file with bracketed name [opportunityId].tsx?

Reply:
Yes, exactly. Since this the name of this file contains a bracketed variable name, that part of the url will become a variable in the query object :smile: